include/osv/semaphore.hh

```diff
@@ -15,6 +15,7 @@
 class semaphore {
 public:
     explicit semaphore(unsigned val);
+    semaphore(int units, int refs, bool named);
```

Comment:
If I understand correctly, in this version of the patch you don't need to change include/osv/semaphore.hh. And also not semaphore.cc by the way.